Figure out why Kotlin packaging is breaking now

https://github.com/marmot-protocol/mdk/actions/runs/22131598401/job/63974132982

```
Run # Build OpenSSL from source for each Android ABI
  # Build OpenSSL from source for each Android ABI
  OPENSSL_ANDROID_DIR="${HOME}/openssl-android"
  mkdir -p "${OPENSSL_ANDROID_DIR}"
  
  for ABI in arm64-v8a armeabi-v7a x86_64; do
    echo "Building OpenSSL for ${ABI}..."
    ./scripts/build-openssl-android.sh "${ABI}" "${OPENSSL_ANDROID_DIR}/${ABI}"
  done
  
  echo "openssl-dir=${OPENSSL_ANDROID_DIR}" >> $GITHUB_OUTPUT
  echo "OpenSSL for Android installed at: ${OPENSSL_ANDROID_DIR}"
  ls -la "${OPENSSL_ANDROID_DIR}"
  shell: /usr/bin/bash -e {0}
  env:
    CARGO_TERM_COLOR: always
    SWIFT_PACKAGE_REPO: marmot-protocol/mdk-swift
    PYTHON_PACKAGE_REPO: marmot-protocol/mdk-python
    RUBY_PACKAGE_REPO: marmot-protocol/mdk-ruby
    KOTLIN_PACKAGE_REPO: marmot-protocol/mdk-kotlin
    CARGO_HOME: /home/runner/.cargo
    CARGO_INCREMENTAL: 0
    CACHE_ON_FAILURE: false
    JAVA_HOME: /opt/hostedtoolcache/Java_Temurin-Hotspot_jdk/17.0.18-8/x64
    JAVA_HOME_17_X64: /opt/hostedtoolcache/Java_Temurin-Hotspot_jdk/17.0.18-8/x64
    ANDROID_HOME: /usr/local/lib/android/sdk
    ANDROID_SDK_ROOT: /usr/local/lib/android/sdk
    NDK_HOME: /home/runner/.setup-ndk/r27d
Building OpenSSL for arm64-v8a...
Building OpenSSL 3.3.2 for arm64-v8a...
  Target: android-arm64
  NDK: /home/runner/.setup-ndk/r27d
  Output: /home/runner/openssl-android/arm64-v8a

Failure!  build file wasn't produced.
Please read INSTALL.md and associated NOTES-* files.  You may also have to
look over your available compiler tool chain or change your configuration.

no NDK aarch64-linux-android-gcc on $PATH at (eval 10) line 143.
Configuring OpenSSL version 3.3.2 for target android-arm64
Using os-specific seed configuration
Error: Process completed with exit code 2.
```

--- a/scripts/build-openssl-android.sh
+++ b/scripts/build-openssl-android.sh
@@ -27,23 +27,27 @@ if [ -z "${NDK_HOME:-}" ]; then
     exit 1
 fi
 
-# Map ABI to OpenSSL target
+# Map ABI to OpenSSL target and NDK clang triple
 case "$ABI" in
     arm64-v8a)
         OPENSSL_TARGET="android-arm64"
         ANDROID_API=21
+        CLANG_TRIPLE="aarch64-linux-android"
         ;;
     armeabi-v7a)
         OPENSSL_TARGET="android-arm"
         ANDROID_API=21
+        CLANG_TRIPLE="armv7a-linux-androideabi"
         ;;
     x86_64)
         OPENSSL_TARGET="android-x86_64"
         ANDROID_API=21
+        CLANG_TRIPLE="x86_64-linux-android"
         ;;
     x86)
         OPENSSL_TARGET="android-x86"
         ANDROID_API=21
+        CLANG_TRIPLE="i686-linux-android"
         ;;
     *)
         echo "Error: Unknown ABI: $ABI"
@@ -70,6 +74,13 @@ esac
 TOOLCHAIN="${NDK_HOME}/toolchains/llvm/prebuilt/${NDK_HOST}"
 export PATH="${TOOLCHAIN}/bin:$PATH"
 export ANDROID_NDK_ROOT="${NDK_HOME}"
+
+# NDK r23+ removed standalone GCC toolchains. OpenSSL's android-* targets
+# look for <triple>-gcc by default, so we explicitly set CC/AR/RANLIB to
+# point at the LLVM/Clang equivalents shipped with modern NDKs.
+export CC="${CLANG_TRIPLE}${ANDROID_API}-clang"
+export AR="llvm-ar"
+export RANLIB="llvm-ranlib"
 
 # Create temporary build directory
 BUILD_DIR=$(mktemp -d)

